### DISCUSSION TITLE
Remote 'default-conan-local' username

### DISCUSSION THREAD

**Discussion by ponponon** [milvus-io/milvus#26156] [Q&A and General discussion]
```shell
git clone git@github.com:milvus-io/milvus.git
cd milvus
pip install conan==1.58.0
make build
```

然后要我输入账号？

```
Remote 'default-conan-local' username:

```

完整的日志:

```shell
libelf/0.8.13: Downloaded recipe revision 4f70a3555809ae50cc8add44f0f84288
libiberty/9.1.0: Not found in local cache, looking in remotes...
libiberty/9.1.0: Trying with 'conancenter'...
Downloading conanmanifest.txt completed [0.10k]                                          
Downloading conanfile.py completed [3.00k]                                               
Downloading conan_export.tgz completed [0.31k]                                           
Decompressing conan_export.tgz completed [0.00k]                                         
libiberty/9.1.0: Downloaded recipe revision a6a0b042484db9cefa233a8aa31a33c5
libsodium/cci.20220430: Not found in local cache, looking in remotes...
libsodium/cci.20220430: Trying with 'conancenter'...
Downloading conanmanifest.txt completed [0.19k]                                          
Downloading conanfile.py completed [7.51k]                                               
Downloading conan_export.tgz completed [0.43k]                                           
Decompressing conan_export.tgz completed [0.00k]                                         
libsodium/cci.20220430: Downloaded recipe revision cd827332735a7278432659372a1efb77
prometheus-cpp/1.1.0: Not found in local cache, looking in remotes...
prometheus-cpp/1.1.0: Trying with 'conancenter'...
Downloading conanmanifest.txt completed [0.10k]                                          
Downloading conanfile.py completed [6.70k]                                               
Downloading conan_export.tgz completed [0.32k]                                           
Decompressing conan_export.tgz completed [0.00k]                                         
prometheus-cpp/1.1.0: Downloaded recipe revision c6c3bf8a1a108fdaf36929ccd22e9082
WARN: prometheus-cpp/1.1.0: requirement libcurl/8.0.1 overridden by your conanfile to libcurl/7.86.0 
folly/2023.07.12@milvus/dev: Not found in local cache, looking in remotes...
folly/2023.07.12@milvus/dev: Trying with 'conancenter'...
folly/2023.07.12@milvus/dev: Trying with 'default-conan-local'...
Please log in to "default-conan-local" to perform this action. Execute "conan user" command.
Remote 'default-conan-local' username: 
```

**Comment by jiaoew1991** [comment]
@ponponon Milvus relies on some fork versions of packages, which can be downloaded from the public repository at https://milvus01.jfrog.io/ui/repos/tree/General/default-conan-local. No username or password is required for accessing this repository. 🤔

**Comment by leozy255** [comment]
How to solve this problem

**Reply by yhmo** [reply]
You can use the command "conan user" to show the user names. My username of 'default-conan-local' is an empty string:
<img width="625" height="62" alt="Screenshot from 2025-11-20 16-16-59" src="https://github.com/user-attachments/assets/86661240-7492-47e8-b3bb-f06a613af157" />

If your user name of 'default-conan-local' is not empty, maybe you have configured it when you installed Conan and you have forgotten it.

**Comment by Shagga01** [comment]
Hi, I am Humphrey O. Founder and Team Leader of Red Waters Group. 
Red Waters Group is assembling a high-caliber global team of innovators, builders, and digital strategists.
We’re creating a future-forward ecosystem where AI, cloud engineering, automation, cybersecurity, product innovation, and digital architecture converge. If you’re a skilled professional ready to build revolutionary systems, contribute your expertise, and grow with a team driven by ambition and impact—we want you with us. Join us as we engineer the next generation of digital solutions.
👉 Apply here: https://forms.gle/RgLpGKKn5MZMVbRq9

**Comment by cyber93** [comment]
**Permission Denied for 'folly' Package in 'default-conan-local' Repository**


**Description**
I was able to build Milvus 2.6.15 successfully until two days ago. However, starting from yesterday, I am encountering the same build failure. It appears to be a permission configuration issue with the default-conan-local package repository.

1. Environment / conanfile.py
The error occurs when trying to pull the following specific revision:
"folly/2023.10.30.08@milvus/dev#81d7729cd4013a1b708af3340a3b04d9"

2. Build Error Log
folly/2023.10.30.08@milvus/dev: Retrieving from server 'default-conan-local' 
folly/2023.10.30.08@milvus/dev: Trying with 'default-conan-local'... 
ERROR: Failed requirement 'folly/2023.10.30.08@milvus/dev' from 'conanfile.py' 
ERROR: Permission denied for user: 'anonymous'. [Remote: default-conan-local]

3. Observed Behavior via Browser
Accessing the repository via the following URL shows a permission-related restriction for anonymous access:
https://milvus01.jfrog.io/ui/repos/tree/General/default-conan-local/milvus/folly/2023.10.30.08/dev/81d7729cd4013a1b708af3340a3b04d9
<img width="1751" height="775" alt="화면 캡처 2026-05-15 144833" src="https://github.com/user-attachments/assets/ee50a848-5f74-4116-9711-387410a81525" />

4. Temporary Workaround and Secondary Issue
If I modify conanfile.py to use a different version as follows:
Change to: "folly/2026.04.20.00"
The permission error is resolved, but the build fails at a later stage due to other dependencies or compatibility issues.

**Reply by yhmo** [reply]
The default-conan-local indeed allows anonymous access.
I am also using anonymous to log in conan, anonymous can access the default-conan-local for folly recipe.
<img width="1031" height="255" alt="Screenshot from 2026-05-15 14-59-13" src="https://github.com/user-attachments/assets/4a51267c-ee1d-4d2f-bab7-7b3ca551daa5" />

Note: we only provide recipes on default-conan-local, no pre-built packages are uploaded, when you build milvus, all the dependent recipes are downloaded to local cache and built locally.

**Reply by cyber93** [reply]
It's asking me to log in again. Could there be a reason for this?
---------------------------------------------------------------------------
Intel1:~/Go_Workspace/milvus-2.6.15$ conan --version
Conan version 1.64.1

Intel1:~/Go_Workspace/milvus-2.6.15$ conan remote list
conancenter: https://center.conan.io [Verify SSL: True]
default-conan-local: https://milvus01.jfrog.io/artifactory/api/conan/default-conan-local [Verify SSL: True]

Intel1:~/Go_Workspace/milvus-2.6.15$ conan user -r default-conan-local
Current user of remote 'default-conan-local' set to: 'None' (anonymous)

Intel1:~/Go_Workspace/milvus-2.6.15$ conan search folly/2023.10.30.08@milvus/dev -r default-conan-local
**Please log in to "default-conan-local" to perform this action. Execute "conan user" command.
Remote 'default-conan-local' username:**

**Reply by yhmo** [reply]
You can try these commands. They remove folly recipe from the local cache to force Conan to contact the remote again:
```
  conan user -c
  conan user -r default-conan-local

  conan remove folly/2023.10.30.08@milvus/dev -c
  CONAN_NON_INTERACTIVE=1 conan search folly/2023.10.30.08@milvus/dev -r default-conan-local
```

Another possible reason is: different network/VPN/proxy path to JFrog.
